### ISSUE TITLE
alfresco/core/ObjectProcessingMixin processObject function converts array to string when it is inside an object

### ISSUE THREAD

**Issue by johhnywishwash** [Alfresco/Aikau#1059]
I have come across an issue where the **_processObject_** function in **_alfresco/core/ObjectProcessingMixin_** will convert a property with an array as its value into a string.  It seems that when processing an object against the currentItem the **_processObject_** function will take the value in the currentItem: 

```
{
    property : ["A","B","C"]
}
```

and replace the token with the array.
If the currentItem is structured so the property is inside in an object

```
{
    object : {
             property : ["A","B","C"]
    }
}
```

then it converts the property into a string and replaces the token with a string.

I have created an example of this issue against version 1.0.67.  I have created an example similar to the **_ObjectProcessingMixinButton_** test.

The page config is : 

```
model.jsonModel = {
	widgets : [ {
		id : "TEST_REPLACE_TOKEN_WITH_ARRAY_IN_OBJECT",
		name : "test/ObjectProcessingMixinButton",
		config : {
			label : "foo.bar",
			token : "{foo.bar}"
		}
	}, {
		id : "TEST_REPLACE_TOKEN_WITH_ARRAY",
		name : "test/ObjectProcessingMixinButton",
		config : {
			label : "bar",
			token : "{bar}"
		}
	}, {
		name : "alfresco/logging/DebugLog"
	} ],
	services : [ ]
};
```

And the test button widget code : 

```
define([ "alfresco/buttons/AlfButton", "alfresco/core/ObjectProcessingMixin",
		"dojo/_base/declare" ], function(AlfButton, ObjectProcessingMixin,
		declare) {

	return declare([ AlfButton, ObjectProcessingMixin ], {

		token : "",

		currentItem : null,

		label : "TEST",

		postMixInProperties : function() {
			this.inherited(arguments);
		},

		onClick : function alfresco_buttons_AlfButton__onClick(evt) {

			this.currentItem = {
				foo : {
					bar : [ "A", "B", "C" ]
				},
				bar : [ "A", "B", "C" ]
			};

			// Publish generated payload
			var payload = this.processObject([ "processCurrentItemTokens" ],
					this._createObject());
			this.alfPublish("GENERATED", payload, true);

			// Stop the event
			if (evt) {
				evt.stopPropagation();
			}
			
		},

		_createObject : function() {
			var obj1 = {
				foo : {
					bar : this.token
				}
			};
			return obj1;
		}
	});
});
```

**Comment by AFaust** [comment]
The real problem is in the **processTokens** function of the **ObjectProcessingMixin**. It doesn't handle dot-notation when checking if the whole string is a token and relegates the handling to the dojo/_base/lang module and its **replace** operation, which always results in a string value.
This behaviour is one that I have addressed in our internal fork of Aikau and I'll create+link a PR for that.

**Comment by draperd** [comment]
Apologies for the delay in responding to this issue - the Aikau team were on vacation, we will start looking through the backlog of issues from today - thanks for your patience!

**Comment by draperd** [comment]
Although there is a proposed fix provided in #1061 it does not come with the required unit tests - therefore I've raised https://issues.alfresco.com/jira/browse/AKU-971 to cover this issue. As this is a bug it will be prioritized for the next sprint. We will add the necessary unit tests to accompany the fix provided, but we can't just merge code with out the necessary accompanying tests.

**Comment by draperd** [comment]
The fix for this is now available in Aikau 1.0.68 that is now available.
